politeiad/cache/cockroachdb/decred.go

```diff
@@ -884,6 +884,243 @@ func (d *decred) cmdTokenInventory(payload string) (string, error) {
 	return string(reply), nil
 }
 
+// getAuthorizeVotesForRecords looks up vote authorizations in the cache for
+// a set of records.
+func (d *decred) getAuthorizeVotesForRecords(recordsMap map[string]Record) (map[string]AuthorizeVote, error) {
+
+	avMap := make(map[string]AuthorizeVote)
+
+	if len(recordsMap) == 0 {
+		return avMap, nil
+	}
+
+	keys := make([]string, 0, len(recordsMap))
+	for token, record := range recordsMap {
+		keys = append(keys, token+strconv.FormatUint(record.Version, 10))
+	}
+
+	avs := make([]AuthorizeVote, 0, len(keys))
+	err := d.recordsdb.
+		Where("key IN (?)", keys).
+		Find(&avs).
+		Error
+	if err != nil {
+		return nil, err
+	}
+
+	for _, av := range avs {
+		avMap[av.Token] = av
+	}
+
+	return avMap, nil
+}
+
+// getStartVotesForAuthorizeVotes looks up the start votes for records which
+// have been authorized to start voting.
+func (d *decred) getStartVotesForAuthorizeVotes(avMap map[string]AuthorizeVote) (map[string]StartVote, error) {
+	svMap := make(map[string]StartVote)
+
+	if len(avMap) == 0 {
+		return svMap, nil
+	}
+
+	tokens := make([]string, 0, len(avMap))
+	for token := range avMap {
+		tokens = append(tokens, token)
+	}
+
+	svs := make([]StartVote, 0, len(tokens))
+	err := d.recordsdb.
+		Where("token IN (?)", tokens).
+		Preload("Options").
+		Find(&svs).
+		Error
+
+	if err != nil {
+		return nil, err
+	}
+	for _, sv := range svs {
+		svMap[sv.Token] = sv
+	}
+
+	return svMap, nil
+}
+
+// getVoteResultsForStartVotes retrieves vote results for records that have
+// begun the voting process. Results are lazily loaded into this table, so
+// some results will need to be manually looked up in the CastVote table.
+func (d *decred) getVoteResultsForStartVotes(svMap map[string]StartVote) (map[string][]decredplugin.VoteOptionResult, error) {
+
+	resMap := make(map[string][]decredplugin.VoteOptionResult)
+
+	if len(svMap) == 0 {
+		return resMap, nil
+	}
+
+	tokens := make([]string, 0, len(svMap))
+	for token := range svMap {
+		tokens = append(tokens, token)
+	}
+
+	vrs := make([]VoteResults, 0, len(tokens))
+	err := d.recordsdb.
+		Where("token IN (?)", tokens).
+		Preload("Results").
+		Preload("Results.Option").
+		Find(&vrs).
+		Error
+	if err != nil {
+		return nil, err
+	}
+
+	for _, vr := range vrs {
+		resMap[vr.Token] = convertVoteOptionResultsToDecred(vr.Results)
+	}
+
+	return resMap, nil
+}
+
+// lookupResultsForVoteOptions looks in the CastVote to see how many votes
+// each option has received.
+func (d *decred) lookupResultsForVoteOptions(options []VoteOption) ([]decredplugin.VoteOptionResult, error) {
```

Comment:
This code is the same manual lookup code from `cmdVoteSummary`.  If you abstract out code into a function like this can you replace the original code, so in the case the manual lookup code in `cmdVoteSummary`, with a function call instead when appropriate?  This will help prevent duplicate code, make bugs easier to fix, etc.